brushfire-core/src/main/scala/com/stripe/brushfire/AnnotatedTree.scala

```diff
@@ -27,6 +27,22 @@ object LeafNode {
 }
 
 case class AnnotatedTree[K, V, T, A: Semigroup](root: Node[K, V, T, A]) {
+
+  private var sumTargetsCache: Option[T] = None
```

Comment:
Yeah, there are sort of 2 issues. One is that caching it is a bit funky, since, technically, we could run `sumTargets` with a different monoid (eg max vs + vs * vs whatever) and so things would subtly break. OTOH, full-tree traversals are generally bad things that we want to avoid. It may be that we do a full tree traversal, but move the caching somewhere else for `sumTargets`?

With annotations, we have the expectation that they are attached to the trees (and both @NathanHowell and @striation just made annotations strict in the inner nodes, which I agree is the right decision, since you can always get laziness with a wrapper type), but `sumTargets` doesn't quite have this (eg it doesn't return a new tree like the annotation methods do).

In any case we should be very explicit about what is going on with some docs on the `sumTargets` method.